src/nnet3/nnet-utils.cc

```diff
@@ -40,6 +40,169 @@ int32 NumInputNodes(const Nnet &nnet) {
   return ans;
 }
 
+int32 GetInputInterval(const Nnet &nnet, std::string input_name) {
+  const std::vector<std::string> &node_names = nnet.GetNodeNames();
+  int32 interval = -1;
+  for (int32 i = 0; i < nnet.NumNodes(); i++) {
+    const NetworkNode &node = nnet.GetNode(i);
+    if (node.node_type == kDescriptor) {
+      std::ostringstream ostr;
+      node.descriptor.WriteConfig(ostr, node_names);
+      std::vector<std::string> tokens;
+      DescriptorTokenize(ostr.str(), &tokens);
+      tokens.push_back("end of input");
+      const std::string *next_token = &(tokens[0]);
+      GeneralDescriptor *gen_desc = GeneralDescriptor::Parse(node_names,
+                                                             &next_token);
+      if (*next_token != "end of input")
+        KALDI_ERR << "Parsing Descriptor, expected end of input but got "
+                  << "'" <<  *next_token << "'";
+      int32 new_interval =
+          GetInputIntervalInternal(*gen_desc, node_names, input_name);
+      if (new_interval != -1) {
+        if (new_interval == 0) {
+          // if input is used only at t=0 from the arg we are looking at, and
```

Comment:
it would be better if you used braces and put the comments inside the braces for the code they relate to- then it will be clearer what the comments are talking about.